kratos/tests/cpp_tests/geometries/test_nurbs_curve.cpp

```diff
@@ -0,0 +1,300 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Thomas Oberbichler
+//
+
+// System includes
+#include <limits>
+
+// External includes
+
+// Project includes
+#include "testing/testing.h"
+#include "containers/pointer_vector.h"
+#include "geometries/nurbs_curve_geometry.h"
+
+#include "tests/cpp_tests/geometries/test_geometry.h"
+
+namespace Kratos {
+namespace Testing {
+
+typedef Node<3> NodeType;
+
+// /// Factory functions
+//namespace {
+    NurbsCurveGeometry<2, PointerVector<Point>> GenerateReferenceCurve2d()
+    {
+        PointerVector<Point> points;
+
+        points.push_back(Point::Pointer(new Point(0, 0, 0)));
+        points.push_back(Point::Pointer(new Point(3.3333333333333335, 1.6666666666666667, 0)));
+        points.push_back(Point::Pointer(new Point(6.6666666666666661, 3.333333333333333, 0)));
+        points.push_back(Point::Pointer(new Point(10, 5, 0)));
+
+        Vector knot_vector = ZeroVector(6);
+        knot_vector[0] = 0.0;
+        knot_vector[1] = 0.0;
+        knot_vector[2] = 0.0;
+        knot_vector[3] = 11.180339887498949;
+        knot_vector[4] = 11.180339887498949;
+        knot_vector[5] = 11.180339887498949;
+
+        int p = 3;
+
+        auto curve = NurbsCurveGeometry<2, PointerVector<Point>>(points, p, knot_vector);
+
+        return curve;
+    }
+
+    NurbsCurveGeometry<3, PointerVector<NodeType>> GenerateReferenceCurve3d()
+    {
+        PointerVector<NodeType> points;
+
+        points.push_back(NodeType::Pointer(new NodeType(1, 0, -25, -5)));
+        points.push_back(NodeType::Pointer(new NodeType(2, -15, -15, 0)));
+        points.push_back(NodeType::Pointer(new NodeType(3, 5, -5, -3)));
+        points.push_back(NodeType::Pointer(new NodeType(4, 15, -15, 3)));
+        points.push_back(NodeType::Pointer(new NodeType(5, 25, 0, 6)));
+        points.push_back(NodeType::Pointer(new NodeType(6, 15, 15, 6)));
+        points.push_back(NodeType::Pointer(new NodeType(7, -5, -5, -3)));
+        points.push_back(NodeType::Pointer(new NodeType(8, -25, 15, 4)));
```

Comment:
```suggestion
        points.push_back(Kratso::make_shared<NodeType>(8, -25, 15, 4));
```

up-to-date c++ code should not contain `new` and `delete`

Reply:
I was having that before and it was not working, thus, I kept it as it is.